core/src/main/java/org/apache/iceberg/ManifestWriter.java

```diff
@@ -149,13 +150,17 @@ public long length() {
     return writer.length();
   }
 
+  void useSequenceNumber(long sequenceNumber) {
+    this.manifestSequenceNumber = sequenceNumber;
+  }
+
   public ManifestFile toManifestFile() {
     Preconditions.checkState(closed, "Cannot build ManifestFile, writer is not closed");
     // if the minSequenceNumber is null, then no manifests with a sequence number have been written, so the min
     // sequence number is the one that will be assigned when this is committed. pass UNASSIGNED_SEQ to inherit it.
     long minSeqNumber = minSequenceNumber != null ? minSequenceNumber : UNASSIGNED_SEQ;
     return new GenericManifestFile(file.location(), writer.length(), specId, content(),
-        UNASSIGNED_SEQ, minSeqNumber, snapshotId,
+        manifestSequenceNumber, minSeqNumber, snapshotId,
```

Comment:
It is also set sequenceNumber for manifestfile, so i think maybe it is same as https://github.com/apache/iceberg/pull/3204. As @rdblue 
 say in https://github.com/apache/iceberg/pull/3204#issuecomment-962701123 maybe we need one new property like dataSeqnumber in manifestfile.

Reply:
But i also think dataSeqnumber in manifestfile will cause more complex when we read data. And it maybe will cause some problem in version compatibility.

Reply:
I think there is some different between #3204 and this PR. In this PR, the specific seqNum will be set to manifest file, and manifest list file will still got a new seqNum when we commit the snapshot of rewrite. In the result, we could got the incremental seqNum from snapshot and got spercific seqNum form data files because data file will inherit seqNum from menifest file but not from snapshot. Then we can use the seqNum of the data file to verify whether there are deleted files to modify the rewritten data files.

But in #3204, the seqNum of snapshot will override by specific seqNum, and we will got two snapshot which have same seqNum. So that, the monotonicity of snapshot seqNum will be break and we can not recognize which snapshot is the new one because they have same seqNum.

Reply:
@rdblue @Reo-LEI As fllows, it is the manifestFile content, it is also inherit old sequence_number and set it in manifestFile, what get same result as https://github.com/apache/iceberg/pull/3204.
{"manifest_path":"hdfs://ns/group/user/root/meta/hive-temp-table/iceberg_hive_catalog_test1.db/sample35/metadata/ee0fd321-8178-4631-b62b-50d6bb365bc2-m1.avro","manifest_length":6674,"partition_spec_id":0,"content":0,"sequence_number":1837,"min_sequence_number":1839,"added_snapshot_id":5384193052566473584,"added_data_files_count":1,"existing_data_files_count":0,"deleted_data_files_count":0,"added_rows_count":14,"existing_rows_count":0,"deleted_rows_count":0,"partitions":{"array":[]}}

Reply:
> and we will got two snapshot which have same seqNum

@Reo-LEI #3204 will not got two snapshot which have same seqNum, but just like this PR get old seqNum in manifestFile.

Reply:
I think there may be some confusion here. This PR should be updated so that the manifest list and manifest files use a newly assigned sequence number and just the data file entries use the specified sequence number.

Reply:
@hameizi, the snapshot and manifest list should be written with the latest sequence number so that we can track it. Re-sequenced files can override the sequence number, but we should have correct values in metadata. We will probably add another field in the manifests to inherit it.

Reply:
> This PR should be updated so that the manifest list and manifest files use a newly assigned sequence number and just the data file entries use the specified sequence number.

Yes I think that is the approach used in this PR